mhs/inbound/inbound/request/tests/test_handler.py

```diff
@@ -26,37 +55,87 @@ class TestInboundHandler(tornado.testing.AsyncHTTPTestCase):
         super().setUp()
 
     def get_app(self):
         return tornado.web.Application([
-            (r".*", handler.InboundHandler, dict(callbacks=self.callbacks, party_id=FROM_PARTY_ID))
+            (r".*", handler.InboundHandler, dict(workflows=self.mocked_workflows, state_store=self.state, party_id=FROM_PARTY_ID))
         ])
 
     @unittest.mock.patch.object(message_utilities.MessageUtilities, "get_timestamp")
     @unittest.mock.patch.object(message_utilities.MessageUtilities, "get_uuid")
-    def test_post(self, mock_get_uuid, mock_get_timestamp):
+    def test_receive_ack(self, mock_get_uuid, mock_get_timestamp):
         mock_get_uuid.return_value = "5BB171D4-53B2-4986-90CF-428BE6D157F5"
         mock_get_timestamp.return_value = "2012-03-15T06:51:08Z"
         expected_ack_response = file_utilities.FileUtilities.get_file_string(
             str(self.message_dir / EXPECTED_RESPONSE_FILE))
         request_body = file_utilities.FileUtilities.get_file_string(str(self.message_dir / REQUEST_FILE))
-        mock_callback = unittest.mock.Mock()
-        self.callbacks[REF_TO_MESSAGE_ID] = mock_callback
 
         ack_response = self.fetch("/", method="POST", body=request_body, headers=CONTENT_TYPE_HEADERS)
 
         self.assertEqual(ack_response.code, 200)
         self.assertEqual(ack_response.headers["Content-Type"], "text/xml")
         xml_utilities.XmlUtilities.assert_xml_equal(expected_ack_response, ack_response.body)
-        mock_callback.assert_called_with(EXPECTED_MESSAGE)
 
-    def test_post_no_callback(self):
-        # If there is no callback registered for the message ID the response is in reference to, an HTTP 500 should be
-        # returned.
-        request_body = file_utilities.FileUtilities.get_file_string(str(self.message_dir / REQUEST_FILE))
+    def test_post_unsolicited_message(self):
+        request_body = file_utilities.FileUtilities.get_file_string(str(self.message_dir / UNSOLICITED_REQUEST_FILE))
 
         response = self.fetch("/", method="POST", body=request_body, headers=CONTENT_TYPE_HEADERS)
 
         self.assertEqual(response.code, 500)
+        message = response.body.decode('utf-8')
+        self.assertEqual(
+            message,
+            '<html><title>500: Unknown message reference</title><body>500: Unknown message reference</body></html>')
+
+    def test_correct_workflow(self):
+        request_body = file_utilities.FileUtilities.get_file_string(str(self.message_dir / REQUEST_FILE))
+
+        ack_response = self.fetch("/", method="POST", body=request_body, headers=CONTENT_TYPE_HEADERS)
+        self.mocked_workflows[workflow.ASYNC_EXPRESS].handle_inbound_message.assert_called()
+
+        self.assertEqual(ack_response.code, 200)
+
+
+class TestInboundWorkflow(tornado.testing.AsyncHTTPSTestCase):
+    """A simple integration test for the async response endpoint."""
+
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    message_dir = pathlib.Path(current_dir) / MESSAGES_DIR
```

Comment:
You might be able to do something like:
```suggestion
    message_dir = pathlib.Path(__file__) / '..' / MESSAGES_DIR
```
based on [this SO answer](https://stackoverflow.com/a/50098973)

Reply:
This failed on the pipeline actually so I'll leave it for the time being as it passed previously